client/server/pages/index.js

```diff
@@ -920,6 +920,28 @@ export default function pages() {
 
 	app.set( 'views', __dirname );
 
+	// Limit allowed header sizes
+	app.use( ( req, res, next ) => {
+		const MAX_HEADER_LENGTH = 8192; // 8 KB total for all headers
+		const MAX_USER_AGENT_LENGTH = 1024; // 1 KB for User-Agent header
+
+		const totalHeaderSize = Object.entries( req.headers ).reduce(
+			( sum, [ key, value ] ) => sum + key.length + String( value ).length,
+			0
+		);
```

Comment:
I wonder if this is another vector. Serializing and summing a long header can be slow as well. Also, what if someone sent too many headers? The loop could hang as well.

Maybe we should switch to rawHeaders since it's already an array.

```suggestion
        if ( req.rawHeaders.length > MAX_HEADER_COUNT ) {
             return res.status( 413 ).send( 'Headers too large.' );
        }
		const totalHeaderSize = req.rawHeaders.reduce(
			( sum, value ) => sum + value.length,
			0
		);
```